plugins/nanospell/plugin.js

```diff
@@ -190,7 +190,7 @@
 			// reached the end of block,
 			// so just return what we've walked
 			// of the current word.
-			if (word) return { word: word, range: wordRange };
+			return { word: word, range: wordRange };
```

Comment:
Yes it is a reversion.  What I was trying to prevent by the `if` check was unnecessarily returning an empty word `""`.  However, if the word currently is a blank word, you don't trigger a return, so it continues through the loop (at least in the case of the top two returns - the bottom return is because you hit the end of the text node anyway, so a change there doesn't make a difference).  Since you are continuing, the text node range offset does not reset so subsequent returns of non-blank words can include spaces and/or multiple words.  This affects words that are sent in the AJAX call, as well as `span` wrapping by `markTyposInRange`, since both have their data supplied by the `WordWalker`